packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1585,11 +1596,35 @@ export class TransactionController extends BaseControllerV1<
    * Ensure `releaseLock` is called once processing of the `nonce` value is complete.
    *
    * @param address - The hex string address for the transaction.
+   * @param networkClientId
    * @returns object with the `nextNonce` `nonceDetails`, and the releaseLock.
    */
-  async getNonceLock(address: string): Promise<NonceLock> {
-    // NOTE(JL): i think this should take in chainId, but not sure how to deal with networkClientId mapping
-    return this.nonceTracker.getNonceLock(address);
+  async getNonceLock(
+    address: string,
+    networkClientId?: NetworkClientId,
+  ): Promise<NonceLock> {
+    let nonceMutexForChainId: Mutex | undefined;
+    if (networkClientId) {
+      const networkClient = this.getNetworkClientById(networkClientId);
+      nonceMutexForChainId = this.nonceMutexByChainId.get(
+        networkClient.configuration.chainId,
+      );
+    }
+
+    const releaseLockForChainId = await nonceMutexForChainId?.acquire();
+    try {
+      const nonceLock = await this.nonceTracker.getNonceLock(address);
```

Comment:
I believe we'll want to get the nonceLock from the correct nonceTracker for the networkClientId here. I realize we haven't yet instantiated the trackingSet so this isn't possible yet.